compiler/qsc_parse/src/item.rs

```diff
@@ -518,3 +525,202 @@ pub(super) fn check_input_parens(inputs: &Pat) -> Result<()> {
         Err(Error(ErrorKind::MissingParens(inputs.span)))
     }
 }
+
+/// Parses an export statement. Exports start with the `export` keyword, followed by a curly brace
+/// and a list of items.
+///
+/// ```qsharp
+/// export {
+///     Foo,
+///     Bar.Baz,
+///     Bar.Quux as Corge
+/// };
+/// ```
+fn parse_export(s: &mut ParserContext) -> Result<ExportDecl> {
+    let lo = s.peek().span.lo;
+    let _doc = parse_doc(s);
+    token(s, TokenKind::Keyword(Keyword::Export))?;
+    token(s, TokenKind::Open(Delim::Brace))?;
+    let items = comma_separated_seq(s, parse_export_item)?;
+    token(s, TokenKind::Close(Delim::Brace))?;
+    token(s, TokenKind::Semi)?;
+
+    Ok(ExportDecl {
+        span: s.span(lo),
+        items: items.0.into(),
+        id: NodeId::default(),
+    })
+}
+
+/// returns the path and the alias, if any
+fn parse_export_item(s: &mut ParserContext) -> Result<ExportItem> {
+    let path = *(path(s)?);
+    let alias = if token(s, TokenKind::Keyword(Keyword::As)).is_ok() {
+        Some(*(ident(s)?))
+    } else {
+        None
+    };
+    Ok(ExportItem { path, alias })
+}
+
+/// Parses import items. Note that import items in Q# can be nested and are a tree structure.
+/// For example:
+/// ```qsharp
+/// import Foo.Bar;
+/// import Foo.{Baz, Quux};
+/// import Foo.{Bar.{Baz, Quux}, Corge as Grault};
+/// ```
+fn parse_import(s: &mut ParserContext) -> Result<ImportDecl> {
+    let lo = s.peek().span.lo;
+    let _doc = parse_doc(s);
+    token(s, TokenKind::Keyword(Keyword::Import))?;
+    let (base_path, _) = seq(s, ident, TokenKind::Dot)?;
```

Comment:
Is a `seq` parser with a `.` separator meaningfully different than a `Path` parser? Why go through the trouble of making `seq` generic instead of using `Path`?

`seq` allows an optional trailing separator. Are all these syntaxes allowed on purpose?

```qsharp
import Foo.{Bar, Bar2};
import Foo{Bar, Bar2};
import Foo.;
```